[Feature] component 페이지 마크업 (avatar, checkbox, switch, header)

soulchicken opened the pull request:
## 🎉 변경 사항
- wow-docs의 component 페이지 중에서 avatar, checkbox, switch, header 페이지를 구현했습니다.
- 간단한 작업들이라 명절이 되기 전에 와다다 했습니다... 페이지가 4개나 되기 때문에 두 분의 리뷰는 필요할거같아요...!

### 🙏 여기는 꼭 봐주세요!
- 페이지별로 있는 `componentTab`에서 지금 당장 구현하기 어려운 부분은 이미지 파일로 덮어놓은 상태입니다.
  - checkbox의 box, switch의 btn 입니다. 그 이유는 hover, pressed와 같은 시점으로 컴포넌트를 그대로 멈춰놓을 방법을 찾지 못했고 구현하기 위해서는 디자인시스템에 제작된 컴포넌트를 수정하거나 새로운 땜빵용 컴포넌트를 만들어야할 것 같아서 이 부분을 회의때 논의하고 반영할 생각이예요.
- header 페이지의 `GuidelineTab`에서 아직 GDSC -> GDGoC로 되지 않은 상태입니다. 디자인팀께 요청드린 상태라서 피그마 바뀌면 업데이트하겠습니다.
- header 페이지의 PC 환경에서의 header는 988px이지만 저희 wow-docs에서는 훨씬 작은 사이즈라서 좌우 여백이 예쁘지 않게 들어간 상태입니다. 이 또한 회의때 논의거리로 가져가겠습니다.

soulchicken committed:
feat: avatar guide svg 이미지

soulchicken committed:
feat: avatar 컴포넌트 페이지 마크업

soulchicken committed:
feat: avatar 페이지 오타 수정

soulchicken committed:
feat: header 페이지 마크업

soulchicken committed:
feat: header guideline 컴포넌트

soulchicken committed:
feat: header componentTab 컴포넌트

soulchicken committed:
feat: switch 페이지 마크업

soulchicken committed:
feat: switch GuidelineTab 컴포넌트

soulchicken committed:
feat: switch ComponentTab 컴포넌트

soulchicken committed:
fix: 폴더명, alt text 변경

soulchicken committed:
feat: checkbox component 페이지의 이미지 삽입

soulchicken committed:
feat: checkbox component 페이지 마크업

soulchicken committed:
feat: checkbox guideline 컴포넌트

soulchicken committed:
feat: checkbox componentTab 컴포넌트

soulchicken committed:
fix: color, spacing, typo의 metadata 부분 수정

SeieunYoo reviewed (commented):
마크업 작업하시느라 수고 하셨습니다!!

Checkbox 와 Switch pressed hover 관리 여부 같은 경우에는 data 속성을 전달하면 컴포넌트를 새로 만들지 않아도 컴포넌트단 제어는 가능할 거 같아요~ 

- Checkbox
checkbox 컴포넌트의 pressed 의 상태는 Checkbox 내부에 input 태그에 data-pressed 속성을 주면 pressed 상태를 표현할 수 있는데요, 
현재 checkbox input 에 접근하기 위해서는 ref 를 전달해서 ref.current 로 요소를 가져와서 data attribute 를 전달할 수 있을 거 같아요.

```ts
 const inputRef = useRef<HTMLInputElement>(null);

  useEffect(() => {
    if (inputRef.current) {
      inputRef.current.setAttribute("data-pressed", "true");
    }
  }, []);
```
- Switch
Switch 컴포넌트는 label 태그에 data 속성을 전달해야 하는데요, 현재 ref 가 input 태그에 전달되고 있어서 label 에 접근하기는 어려울 거 같아요. 🤔
`document.querySelector` 로 label 요소를 가져와야될 듯 한데 아래 방법처럼 돔에 접근하는 방식으로 구현해보는 건 어떨까요? 
```ts
 useEffect(() => {
   //label for 속성으로 요소 가져오기
    const labelElement = document.querySelector('label[for="example id"]');
    if (labelElement) {
      labelElement.setAttribute("data-hover", "true");
    }
  }, []);

<Switch inputProps={{ id: "example id" }} />
```

soulchicken committed:
feat: 코드리뷰 반영하여 checkbox 컴포넌트 탭 수정

- pressed, checked, disabled에 맞게 구현
- 코드리뷰에 나온 내용을 토대로 useRef를 사용해서 attribute 추가

soulchicken committed:
feat: 코드리뷰 반영하여 SwitchBtn Card 수정

- querySelector로 요소를 조작하는 방식을 사용
- disabled의 경우 Switch 컴포넌트가 disabled와 checked가 공존하기 힘들기에 컴포넌트 따로 제작

soulchicken committed:
feat: header 페이지 GDSC에서 GDG로 변경

- GDGoC로 바뀐 내용 반영
- 디자이너님과 이야기하여 PC, Mobile 순서 변경

soulchicken committed:
refactor: checkbox componentTab 상수화, map 순회

eugene028 commented on apps/wow-docs/app/component/switch/_components/ComponentTab.tsx:
```diff
@@ -0,0 +1,43 @@
+          height={188}
+          src="/component/switch/switch-component-btn.svg"
+          width={166}
+        />
```
P1;
요 부분 이미지가 깨지는 것 같은데 확인해주실 수 있으신가용?
![스크린샷 2025-01-28 오후 4 24 17](https://github.com/user-attachments/assets/17c63b13-5da2-4bd2-a8bf-8ddd767243b2)

soulchicken replied:
옙 제거했습니당🔥 [fix: 사용하지 않는 이미지 태그 제거](https://github.com/GDSC-Hongik/wow-design-system/pull/197/commits/405651342a27cdf3739448b4960b17be402123d3)

soulchicken committed:
fix: 사용하지 않는 이미지 태그 제거

SeieunYoo reviewed (approved):
Lgtm👍

soulchicken committed:
Merge branch 'main' into feature/avatar-switch-ect-page
